data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessor.java

```diff
@@ -46,23 +70,50 @@ public GeoIPProcessor(PluginSetting pluginSetting,
   @Override
   public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
 
-    //TODO : logic call the enrichment of data class methods
-    return null;
+    Map<String, Object> geoData;
+
+    for (final Record<Event> eventRecord : records) {
+      Event event = eventRecord.getData();
+      for (KeysConfig key : geoIPProcessorConfig.getKeysConfig()) {
+        String source = key.getKeyConfig().getSource();
+        List<String> attributes = key.getKeyConfig().getAttributes();
+        String ipAddress = event.get(source, String.class);
+
+        //Lookup from DB
+        if (ipAddress != null && (!(ipAddress.isEmpty()))) {
+          try {
+            if (!(IPValidationcheck.ipValidationcheck(ipAddress))) {
```

Comment:
OK, I see. `IPValidationcheck` returns false if it's public. This is a little counterintuitive. Since you are validating it, it should return true if it's valid (i.e. public here).

Reply:
Modified the code to check the IP is public or not. Now the if condition check is as below:
if (IPValidationcheck.isPublicIpAddress(ipAddress))